pupil_src/shared_modules/camera_models.py

```diff
@@ -462,10 +482,10 @@ def __init__(self, K, D, resolution, name):
         self.name = name
 
     def update_camera_matrix(self, camera_matrix):
-        self.K[:] = np.array(camera_matrix)
+        self.K = np.array(camera_matrix).reshape(self.K.shape)
 
     def update_dist_coefs(self, dist_coefs):
-        self.D[:] = np.array(dist_coefs)
+        self.D = np.array(dist_coefs).reshape(self.D.shape)
```

Comment:
Please use np.asanyarray() here, in order to avoid as much overhead as possible, if a numpy array was passed already. Please apply the same change to update_camera_matrix